MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffFittingPresenter.cpp

```diff
@@ -348,15 +359,21 @@ void EnggDiffFittingPresenter::processMultiRun(
 void EnggDiffFittingPresenter::processSingleRun(
     const std::string &focusDir, const std::string &strFocusedFile,
     std::vector<std::string> &runnoDirVector,
-    const std::vector<std::string> &splitBaseName,
-    std::vector<std::string> &runNoVec) {
+    const std::vector<std::string> &splitBaseName) {
+
+  if (g_fitting_runno_counter == 0) {
+    g_multi_run_directories.clear();
+  }
 
   // to track the FittingRunnoChanged loop number
   g_fitting_runno_counter++;
 
   m_view->setFittingSingleRunMode(true);
 
-  updateFittingDirVec(focusDir, strFocusedFile, runnoDirVector);
+  // dummy vector will not be used
+
+  std::vector<std::string> foundRunNumber;
+  updateFittingDirVec(focusDir, strFocusedFile, runnoDirVector, foundRunNumber);
```

Comment:
We could use a default parameter however I'm trying to think of a way we can hint to the compiler not to create this variable as it won't be used or if it will be smart enough to figure that out